arrayService.spec.js

```diff
@@ -0,0 +1,359 @@
+            assert.isTrue(arrayService.swapNodes("0", "1"));
+        });
+
+    });
+    
+    describe("sortStep", function() {
+       
+       var elem0, elem1, elem2, elem3, elem4;
+       beforeEach(function() {
+           elem0 = {
+                id: "node0",
+                value: 0
+           };
+           elem1 = {
+                id: "node1",
+                value: 100
+           };
+           elem2 = {
+                id: "node2",
+                value: -17
+           };
+           elem3 = {
+                id: "node3",
+                value: 3.14
+           };
+            elem4 = {
+                id: "node4",
+                value: 7
+            };
+       });
+       
+       it("makes active two first elems", function(){
+            var expectedResult = {
+                activeNodeIds: [elem0.id, elem1.id],
+                newStableNodeIds: [],
+                prevNodeIds: [elem4.id]
+            };
+            
+            arrayService.initArray([elem0, elem1, elem2, elem3, elem4]);
+            var actualResult = arrayService.sortStep();
+            assert.equal(expectedResult.activeNodeIds.toString(), actualResult.activeNodeIds.toString());
+            assert.equal(expectedResult.newStableNodeIds.toString(), actualResult.newStableNodeIds.toString());
+            assert.equal(expectedResult.prevNodeIds.toString(), actualResult.prevNodeIds
+            .toString());
+       });
+       
+       it("makes active two last elems", function(){
+            var expectedResult = {
+                activeNodeIds: [elem3.id, elem4.id],
+                newStableNodeIds: [],
+                prevNodeIds: [elem3.id, elem2.id]
+            };
+            
+            arrayService.initArray([elem0, elem1, elem2, elem3, elem4]);
+            arrayService.sortStep();
+            arrayService.sortStep();
+            arrayService.sortStep();
+            var actualResult = arrayService.sortStep();
+            assert.equal(expectedResult.activeNodeIds.toString(), actualResult.activeNodeIds.toString());
+            assert.equal(expectedResult.newStableNodeIds.toString(), actualResult.newStableNodeIds.toString());
+            assert.equal(expectedResult.prevNodeIds.toString(), actualResult.prevNodeIds
+            .toString());
+       });
+       
+       it("makes stable last elem", function(){
+            var expectedResult = {
+                activeNodeIds: [],
+                newStableNodeIds: [elem4.id],
+                prevNodeIds: [elem4.id, elem3.id]
+            };
+            
+            arrayService.initArray([elem0, elem1, elem2, elem3, elem4]);
+            arrayService.sortStep();
+            arrayService.sortStep();
+            arrayService.sortStep();
+            arrayService.sortStep();
+            var actualResult = arrayService.sortStep();
+            assert.equal(expectedResult.activeNodeIds.toString(), actualResult.activeNodeIds.toString());
```

Comment:
Array elements also possible to check by sameMembers : assert.sameMembers(expectedResult.newStableNodeIds, actualResult.newStableNodeIds);